sycl/source/detail/global_handler.cpp

```diff
@@ -28,17 +28,24 @@ namespace sycl {
 __SYCL_INLINE_VER_NAMESPACE(_V1) {
 namespace detail {
 
+using LockGuard = std::lock_guard<SpinLock>;
+
+GlobalHandler *GlobalHandler::MSyclGlobalObjectsHandler = new GlobalHandler();
```

Comment:
I'm not sure when this initialization really happens. Is it possible that somebody calls `GlobalHandler::instance` before this initialization takes place? Probably, it would be more safe approach to have an `std::call_once` in the `GlobalHandler::instance`.